MagneticField/GeomBuilder/src/BaseVolumeHandle.h

```diff
@@ -0,0 +1,219 @@
+
+    /// Return the center of the volume
+    const GlobalPoint& center() const;
+
+    /// Distance of (x,y) plane from origin
+    const double RN() const { return theRN; }
+
+    /// Get the current surface on specified side.
+    const Surface& surface(int which_side) const;
+
+    const Surface& surface(Sides which_side) const;
+
+    /// Find out if two surfaces are the same physical surface
+    bool sameSurface(const Surface& s1, Sides which_side, float tolerance = 0.01);
+
+    /// Assign a shared surface perorming sanity checks.
+    bool setSurface(Surface& s1, Sides which_side);
+
+    /// if the specified surface has been matched.
+    bool isPlaneMatched(int which_side) const { return isAssigned[which_side]; }
+
+    int references(int which_side) const {  // FIXME!
+      /*     return surfaces[which_side]->references(); */
+      return 0;
+    }
+
+    /// Name of the volume
+    std::string name;
+
+    /// Name of magnetic field table file
+    std::string magFile;
+
+    /// volume number
+    unsigned short volumeno;
+
+    /// copy number
+    unsigned short copyno;
+
+    // Phi ranges: Used by: LessDPhiMax; bSector; bSlab::[min|max]Phi();
+    // MagBSector, MagBRod
+
+    /// Minimum value of phi covered by the volume
+    // FIXME: actually returns phi of the point on median plane of the -phi
+    // surface, except for trapezoids where the absoulte min has been implemented
+    Geom::Phi<float> minPhi() const { return thePhiMin; }
+
+    /// Maximum value of phi covered by the volume
+    // FIXME: actually returns phi of the point on median plane of the +phi
+    // surface
+    Geom::Phi<float> maxPhi() const { return surface(SurfaceOrientation::phiplus).position().phi(); }
+
+    /// Z limits.
+    // ASSUMPTION: Computed on median Z plane, but this is not a problem since
+    // all Z planes are orthogonal to the beam line in the current geometry.
+    double minZ() const { return surface(SurfaceOrientation::zminus).position().z(); }
+    double maxZ() const { return surface(SurfaceOrientation::zplus).position().z(); }
+
+    /// Minimum R for any point within the volume
+    double minR() const { return theRMin; }
+
+    /// Position and rotation
+    const GloballyPositioned<float>* placement() const { return refPlane; }
+
+    /// The surfaces and they orientation, as required to build a MagVolume.
+    virtual std::vector<VolumeSide> sides() const = 0;
+
+    /// Pointer to the final MagVolume (must be set from outside)
+    MagVolume6Faces* magVolume;
+
+    bool toExpand() const { return expand; }
+
+    /// Temporary hack to pass information on material. Will eventually be replaced!
+    bool isIron() const { return isIronFlag; }
+
+    /// The sector for which an interpolator for this class of volumes should be built
+    int masterSector;
+
+    /// Shape of the solid
+    virtual DDSolidShape shape() const = 0;
+
+  protected:
+    typedef ReferenceCountingPointer<Surface> RCPS;
```

Comment:
Same as above, cf: https://github.com/cms-sw/cmssw/pull/27674#discussion_r316906238

Reply:
As Nicola mentioned, my previous comment explained the issue. Here's some more detail. The `VolumeSide` constructor takes a non-const pointer to `Surface`. `VolumeSide` is defined in `"MagneticField/VolumeGeometry/interface/VolumeSide.h`, outside the scope of this PR. In the `volumeHandle::sides()` method, `Surface` pointers from the `surfaces` member are passed to the `VolumeSide` constructor. In the old code, `surfaces` was defined to contain const pointers, and a const_cast was used to remove the constness. To avoid the static analzyzer complaint about the const_cast, I made `surfaces` non-const by changing the `RCPS` typedef.
Since the pointers contained in `surfaces` are passed to `VolumeSide`, it seems to make sense to make `surfaces` non-const. However, we could go back to the original approach, which is to keep `surfaces` const until the pointers need to be passed to `VolumeSide`, at which point they must be const_cast to take away the constness.

Let me know which approach we should follow in this PR.

Reply:
> Let me know which approach we should follow in this PR.

from what I found in github cmssw, the preferred solution to me is to make VolumeSide use ConstReferenceCountingPointer and also a ```const Surface*``` in the interface.
It seems like this flexibility for VolumeSide to let the surface be modified was never needed and it leads to awkward code patterns.

I'm fine with simply reverting to the old state using const_cast (it seems to me anyways more acceptable than the current state)

Reply:
I am restoring the old way, including the const_cast. I will leave `VolumeSide` revision to Nicola.